### PR TITLE
fix(Ghost): updateOffer if offerAlreadyExist

### PR DESCRIPTION
**Pull request by peterMangrove**

### CODE REVIEW

File: src/toy_strategies/offer_maker/Ghost.sol [hunk 1]
```diff
@@ -15,6 +15,7 @@ pragma abicoder v2;
 
 import "src/strategies/offer_maker/abstract/Direct.sol";
 import "src/strategies/routers/SimpleRouter.sol";
+import {t_of_struct as packOffer} from "src/preprocessed/MgvOffer.post.sol";
```
Comment (jkrivine) [inline src/toy_strategies/offer_maker/Ghost.sol]: you should not need this

File: src/toy_strategies/offer_maker/Ghost.sol [hunk 2]
```diff
@@ -74,44 +75,59 @@ contract Ghost is Direct {
...
-        gasreq: offerGasreq(),
-        gasprice: 0,
-        pivotId: pivot2,
-        fund: 0,
-        noRevert: false,
-        owner: msg.sender
-      })
-    );
+    (MgvStructs.OfferUnpacked memory offer1, MgvStructs.OfferDetailUnpacked memory offerDetails1) =
+      MGV.offerInfo(address(BASE), address(STABLE1), offerId1);
```
Comment (jkrivine) [inline src/toy_strategies/offer_maker/Ghost.sol]: this is way too gas costly. You want to get the packed offer and offer detail and prune whatever info you want form the packed data.
```solidity
MgvStructs.OfferPacked offer = mgv.offers(address(BASE), address(STABLE1), offerId1);
MgvStructs.OfferDetailPacked detail = mgv.offerDetails(address(BASE), address(STABLE1), offerId1);
```
and then `offer.gives()` or `detail.gasreq()` to extract a specific field

**Review by jkrivine — CHANGES_REQUESTED** [review]

File: src/toy_strategies/offer_maker/Ghost.sol (hunk 1 above)
Comment (peterMangrove) [reply src/toy_strategies/offer_maker/Ghost.sol]: Done

File: src/toy_strategies/offer_maker/Ghost.sol (hunk 2 above)
Comment (peterMangrove) [reply src/toy_strategies/offer_maker/Ghost.sol]: Done

**Review by jkrivine — APPROVED** [review]

**Pull request closed without merging**